### ISSUE TITLE
🐛 [Bug Report] - % damage calculations

### ISSUE THREAD

**Issue by valens14** [Yatzii93/TBC5MAN-Bug-Tracker#94]
# Decription: #
One or possibly multiple talents/debuffs increasing damage done is not calculating multiplicatively causing mage damage dealt to be approximately 92-93% of what it should be with the following list of talents/debuffs:
Improved Scorch
Curse of Elements
Misery
Playing with Fire
Fire Power
Molten Fury

This could also be the case for other damage done modifiers not listed and affect several other classes.

# How it works: #
The following table was performed by using a sample size of 20 fireballs per category to verify that each individual talent/debuff properly raised damage by the correct value.  Fireballs that were partially resisted were neglected from the samples.  Partial resistance was separately verified to be correct and not causing any issues.  Individually each buff/debuff performs as it should.  The last two rows show the data when all buffs/debuffs were combined that shows the observed damage was lower than the calculated damage.  This indicates that one or more buff/debuff are calculating additively instead of multiplicatively.
![data table](https://user-images.githubusercontent.com/6764108/86047039-3dfc0a80-ba1c-11ea-965d-6b57b20cf467.png)
  
The data in this table suggests that it is likely that ALL damage values are being calculated additively since the non crit sample average was 3420 which is very close to the calculated average of 3391 if using additive properties for all talents/debuffs.

# How it should work: #
Example Calculations for a non crit assuming a base average fireball damage of 719 with fire spell power value of 1184.
Incorrect Additive:
Damage = (719 + (1184 * 1.15)) * (1.0 + 0.15 + 0.10 + 0.05 + 0.03 + 0.10 + 0.20) = 3391

Correct Multiplicative:
Damage = (719 + (1184 * 1.15)) * 1.15 * 1.10 * 1.05 * 1.03 * 1.10 * 1.20 = 3757

# Source - You need to provide a source as evidence: #

https://wowwiki.fandom.com/wiki/Percent
"As of Patch 1.12, all modifiers have been changed to multiplicative (as opposed to additive). This means that an effect of +30% and an effect of +50% would not equal an effect of +80%, but instead an effect of +95%."

https://wowwiki.fandom.com/wiki/Fire_Power
"This is a multiplicative talent that is applied after the addition of spell damage item bonuses, and is multiplied together with similar talents, buffs and vulnerabilities."
"This damage multipler stacks with other effects that increase damage by a percentage, such as debuffs like Curse of the Elements and other mage talents like Playing with Fire. These percentages are multiplied together when stacking."

https://wowwiki.fandom.com/wiki/Playing_with_Fire#:~:text=Playing%20with%20Fire%20is%20an,damage%20that%20the%20mage%20causes.
"It is a multiplicative increase which is applied after the addition of spell damage bonuses from items."

https://wowwiki.fandom.com/wiki/Molten_Fury
"Also, because this is a multiplicative damage increase, it increases more effectively as other vulnerabilities are applied to the target (such as Improved Scorch and Curse of the Elements) as well was gaining increases from talents such as Playing with Fire and Fire Power."

# Additional Information: #

**Comment by Yatzii93** [comment]
"The following table was performed by using a sample size of 20 fireballs per category" way to low a sample size to make any meaningful judgement. Any swing in the final overall calcuation either way can easily be accounted for with just how small the sample size is.

**Comment by Yatzii93** [comment]
Plus I'd like to also make the point that patch 1.12 was basically the beginning of the game. It would be insane to think the calculations are additive in the core after so many years. It would also affect every other main core to from vanilla to wotlk considering a lot of their code is the same or similar in regard to general damage calculations. As far as I can see as well it's correct. If there is a problem I'd personally be looking elsewhere.

**Comment by valens14** [comment]
Going to close the issue.  Some more testing layering a few buffs together and doing >100 fireballs produced near exact value for both crits and non crits for the damage being calculated multiplicatively.  I believe I just made an error in calculations when I was manually filtering out partials for those two lines of data.  I think when I averaged it I probably miscounted the number of fireballs I threw producing erronously low results.
